File: OOP/06. CommonTypeSystem/04. Person/Person.cs
using System;

class Person
{
    public string Name { get; private set; }
    public int? Age { get; private set; }

    public Person(string name, int? age = null)
    {
        this.Name = name;
        this.Age = age;
    }

    public override string ToString()
    {
        return string.Format("Name: {0}; Age: {1}",
            this.Name, this.Age == null ? "<unspecified>" : this.Age.ToString());
    }
}

File: Javascript-Apps/01. AdvancedOOP/01. School-ClassicOOP/scripts/Person.js
var Person = Class.create({
    init: function (firstName, lastName, age) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.age = age;
    },

    introduce: function () {
        return "Name: " + this.firstName + " " + this.lastName + ", Age: " + this.age;
    }
});

var Student = Class.create({
    init: function (firstName, lastName, age, grade) {
        this._super.init(firstName, lastName, age);
        this.grade = grade;
    },

    introduce: function () {
        return this._super.introduce() + ", Grade: " + this.grade;
    }
});

var Teacher = Class.create({
    init: function (firstName, lastName, age, speciality) {
        this._super.init(firstName, lastName, age);
        this.speciality = speciality;
    },

    introduce: function () {
        return this._super.introduce() + ", Specialty: " + this.speciality;
    }
});

Student.inherit(Person);
Teacher.inherit(Person);